src/main/java/com/samoilenko/onlinebookstore/validation/constraints/FieldMatchValidator.java

```diff
@@ -0,0 +1,19 @@
+package com.samoilenko.onlinebookstore.validation.constraints;
+
+import com.samoilenko.onlinebookstore.dto.UserRegistrationRequestDto;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class FieldMatchValidator implements
+        ConstraintValidator<FieldMatch, UserRegistrationRequestDto> {
```

Comment:
* first and second fields that you pass to the annotation are not used here
* Annotation is Called `FieldMatch` not a **password** match - we should validate if 2 fields that we received as an arguments in annotation are equal
* use reflection to get the value of those 2 fields
* And it should be applicable not only for `UserRegistrationRequestDto` but for any class
```suggestion
public class FieldMatchValidator implements
        ConstraintValidator<FieldMatch, Object> {
```